stdlib/public/core/ArrayBuffer.swift

```diff
@@ -189,6 +189,10 @@ extension _ArrayBuffer {
   internal __consuming func _consumeAndCreateNew(
     bufferIsUnique: Bool, minimumCapacity: Int, growForAppend: Bool
   ) -> _ArrayBuffer {
+    if !_isNative && capacity >= minimumCapacity,
+       let associatedBuffer = getAssociatedBuffer() {
+      return _ArrayBuffer(_buffer: associatedBuffer, shiftedToStartIndex: 0) 
```

Comment:
This does not look right -- IIUC this function needs to return a uniquely referenced buffer, which (again IIUC) this won't be

Reply:
Oh interesting point. It will *become* uniquely referenced the moment it replaces the old buffer (because deallocating the old buffer will remove the association). Is that sufficient?

Reply:
hmm no it is not, because the old buffer might have an extra reference from another variable

Reply:
Filed a radar to track figuring out how to do this safely, going to just rip it out for now